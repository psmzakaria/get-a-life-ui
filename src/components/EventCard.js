import React from "react";
import { Card, Icon } from "semantic-ui-react";

const EventCard = props => {
  const { title, hostId, description } = props.event;
  const { username: hostName } = hostId;
  return (
    <Card onClick={props.handleOpen} color="violet">
      <Card.Content>
        <Card.Header>{title}</Card.Header>
        <Card.Meta>
          <span>Host: {hostName}</span>
        </Card.Meta>
      </Card.Content>
      <Card.Content extra>{props.status}</Card.Content>
    </Card>
  );
};

export default EventCard;
