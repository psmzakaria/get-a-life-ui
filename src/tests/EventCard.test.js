import React from "react";
import EventCard from "../components/EventCard";

import ShallowRenderer from "react-test-renderer/shallow";

it("renders the create component of the app ", () => {
  const renderer = new ShallowRenderer();
  const event = {
    title: "birthday",
    hostId: {
      username: "lala"
    },
    proposedDates: ["20180813", "20180814", "20180815"]
  };
  const status = "Pending";

  renderer.render(<EventCard event={event} status={status} />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
