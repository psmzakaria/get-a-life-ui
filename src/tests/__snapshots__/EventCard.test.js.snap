// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the create component of the app  1`] = `
<Card
  color="violet"
  onClick={undefined}
>
  <CardContent>
    <CardHeader>
      birthday
    </CardHeader>
    <CardMeta>
      <span>
        Host: 
        lala
      </span>
    </CardMeta>
  </CardContent>
  <CardContent
    extra={true}
  >
    Pending
  </CardContent>
</Card>
`;
